### PR TITLE
feat(EpochBasedInflationaryVoteToken): test inflation from multiple delegatees

### PR DESCRIPTION
**Pull request by PierrickGT**
Issue: https://github.com/sherlock-audit/2023-10-mzero-judging/issues/64

This PR add a test illustrating the scenario outlined by the warden and invalidates the described issue.

### CODE REVIEW

**Review by deluca-mike — CHANGES_REQUESTED** [review]

File: src/abstract/EpochBasedInflationaryVoteToken.sol [hunk 1]
```diff
@@ -277,9 +279,13 @@ abstract contract EpochBasedInflationaryVoteToken is IEpochBasedInflationaryVote
...
 
         uint256 inflatedBalance_ = balance_;
+
+        // NOTE: Only need to check if the latest delegatee has participated,
+        //       since the inflation is realized by syncing in the `_delegate()` function.
         address delegatee_ = _getDelegatee(account_, lastEpoch_); // Internal avoids `_revertIfNotPastTimepoint`.
 
-        // NOTE: Starting from the epoch after the latest sync, before `lastEpoch_`.
+        // NOTE: Starting from the epoch after the latest sync,
+        //       and looping through the subsequent epochs until `lastEpoch_` to get the unrealized inflation.
```
[the remark this answers is not in the document]
Comment (PierrickGT) [reply src/abstract/EpochBasedInflationaryVoteToken.sol]: Fixed in: https://github.com/MZero-Labs/ttg/pull/262/commits/afa804a9e84d422fde84b12cbb19284a025eb420

File: src/abstract/EpochBasedInflationaryVoteToken.sol [hunk 2]
```diff
@@ -277,9 +279,13 @@ abstract contract EpochBasedInflationaryVoteToken is IEpochBasedInflationaryVote
         if (balance_ == 0) return 0; // No inflation if the account had no balance.
 
         uint256 inflatedBalance_ = balance_;
+
+        // NOTE: Only need to check if the latest delegatee has participated,
+        //       since the inflation is realized by syncing in the `_delegate()` function.
```
[the remark this answers is not in the document]
Comment (PierrickGT) [reply src/abstract/EpochBasedInflationaryVoteToken.sol]: Fixed in: https://github.com/MZero-Labs/ttg/pull/262/commits/afa804a9e84d422fde84b12cbb19284a025eb420

File: src/abstract/EpochBasedInflationaryVoteToken.sol [hunk 3]
```diff
@@ -90,6 +90,8 @@ abstract contract EpochBasedInflationaryVoteToken is IEpochBasedInflationaryVote
 
     /**
      * @dev   Delegate voting power from `delegator_` to `newDelegatee_`.
+     * @dev   The inflation is realized for the current delegatee by calling `_sync()`
+     *        before setting `newDelegatee_` as the delegatee.
```
[the remark this answers is not in the document]
Comment (PierrickGT) [reply src/abstract/EpochBasedInflationaryVoteToken.sol]: Fixed in: https://github.com/MZero-Labs/ttg/pull/262/commits/afa804a9e84d422fde84b12cbb19284a025eb420

File: test/Registrar.t.sol [hunk 4]
```diff
@@ -53,7 +53,7 @@ contract RegistrarTests is Test {
         _registrar = new Registrar(address(_zeroGovernor));
     }
 
-    function test_initialState() external view {
```
[the remark this answers is not in the document]
Comment (PierrickGT) [reply test/Registrar.t.sol:56]: It is failing to compile otherwise.

**Review by deluca-mike — APPROVED** [review]

File: test/Registrar.t.sol (hunk 4 above)
[the remark this answers is not in the document]
Comment (deluca-mike) [reply test/Registrar.t.sol:56]: Odd

**Pull request merged**